src/System.Threading/src/System/Threading/ReaderWriterLock.cs

```diff
@@ -0,0 +1,1339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.ConstrainedExecution;
+
+namespace System.Threading
+{
+    public sealed class ReaderWriterLock : CriticalFinalizerObject
+    {
+        private const int InvalidThreadID = -1;
+        private const ushort MaxAcquireCount = ushort.MaxValue;
+        private static readonly int DefaultSpinCount = Environment.ProcessorCount != 1 ? 500 : 0;
+
+        private static long s_mostRecentLockID;
+
+        private ManualResetEvent _readerEvent;
+        private AutoResetEvent _writerEvent;
+        private long _lockID;
+        private volatile int _state;
+        private int _writerID = InvalidThreadID;
+        private int _writerSeqNum;
+        private ushort _writerLevel;
+
+#if RWLOCK_STATISTICS
```

Comment:
No, it looks like it's just there for manual checking. I thought it might have been useful in the past, but it's just stats and not really logging anything. Perhaps I should just remove it?